src/libstd/sys/unix/backtrace/mod.rs

```diff
@@ -0,0 +1,125 @@
+/// the file. This loosely corresponds to pub and priv functions in Rust.
+///
+/// Armed with this knowledge, we know that our solution for address to symbol
+/// translation will need to consult both the local and dynamic symbol tables.
+/// With that in mind, here's our options of translating an address to
+/// a symbol.
+///
+/// * Use dladdr(). The original backtrace()-based idea actually uses dladdr()
+///   behind the scenes to translate, and this is why backtrace() was not used.
+///   Conveniently, this method works fantastically on OSX. It appears dladdr()
+///   uses magic to consult the local symbol table, or we're putting everything
+///   in the dynamic symbol table anyway. Regardless, for OSX, this is the
+///   method used for translation. It's provided by the system and easy to do.o
+///
+///   Sadly, all other systems have a dladdr() implementation that does not
+///   consult the local symbol table. This means that most functions are blank
+///   because they don't have symbols. This means that we need another solution.
+///
+/// * Use unw_get_proc_name(). This is part of the libunwind api (not the
+///   libgcc_s version of the libunwind api), but involves taking a dependency
+///   to libunwind. We may pursue this route in the future if we bundle
+///   libunwind, but libunwind was unwieldy enough that it was not chosen at
+///   this time to provide this functionality.
+///
+/// * Shell out to a utility like `readelf`. Crazy though it may sound, it's a
+///   semi-reasonable solution. The stdlib already knows how to spawn processes,
+///   so in theory it could invoke readelf, parse the output, and consult the
+///   local/dynamic symbol tables from there. This ended up not getting chosen
+///   due to the craziness of the idea plus the advent of the next option.
+///
+/// * Use `libbacktrace`. It turns out that this is a small library bundled in
+///   the gcc repository which provides backtrace and symbol translation
+///   functionality. All we really need from it is the backtrace functionality,
+///   and we only really need this on everything that's not OSX, so this is the
+///   chosen route for now.
+///
+/// In summary, the current situation uses libgcc_s to get a trace of stack
+/// pointers, and we use dladdr() or libbacktrace to translate these addresses
+/// to symbols. This is a bit of a hokey implementation as-is, but it works for
+/// all unix platforms we support right now, so it at least gets the job done.
+
+#[cfg(not(all(target_os = "ios", target_arch = "arm")))]
+pub use sys::backtrace::gcc_s::write;
+
+#[cfg(all(target_os = "ios", target_arch = "arm"))]
+pub use sys::backtrace::libbacktrace::write;
+
+use io;
+use io::prelude::*;
+use libc;
+use str;
+
+use sys_common::backtrace::{demangle, HEX_WIDTH};
+
+// symbol resolvers:
+mod printer;
+
+// tracing impls:
+#[cfg(not(all(target_os = "ios", target_arch = "arm")))] mod gcc_s;
+#[cfg(all(target_os = "ios", target_arch = "arm"))] mod libbacktrace;
```

Comment:
I think this module would be more appropriately named as something like `unix_backtrace` or `backtrace_fn` as `libbacktrace` is different from the `backtrace` function in libc.

Also, can this use `#[path = "..."]` and `mod imp;` for both cases to avoid having the same `#[cfg]` for the import and the module declaration?